examples/common/mcstack/FairStack.cxx

```diff
@@ -395,16 +394,15 @@ TParticle* FairStack::GetParticle(Int_t trackID) const
 
 void FairStack::SelectTracks()
 {
-
     // --> Clear storage map
     fStoreMap.clear();
 
     // loop over tracks created by FastSim to order the tracks
-    for (fFSTrackIter = fFSTrackMap.begin(); fFSTrackIter != fFSTrackMap.end(); ++fFSTrackIter) {
+    for (auto tmentry : fFSTrackMap) {
```

Comment:
```suggestion
    for (const auto [newTrack, originalTrack] : fFSTrackMap) {
```

see https://en.cppreference.com/w/cpp/language/structured_binding

Perhaps you find better names even, `.first`/`.second` is definitely terrible for readability.

Reply:
TL;DR *I* wont do this in this PR.

To Be Honest, I have not yet udnerstood what this map is actually mapping. Yes, it's something "old" / "new". But I really don't understand it yet. So if someone else has this understanding, then I'd ask that person to go ahead and improve things!

Reply:
This comment I took the terms "new" and "original" from:

https://github.com/FairRootGroup/FairRoot/blob/ef00300f8c7b3394619e84b7f1cae98d29133509/base/sim/FairGenericStack.h#L146-L147

Reply:
But I agree, it does not help much.

Reply:
Can we resolve this and postpone for the next refactoring round?

Reply:
I would rather just ask someone?
@MohammadAlTurany @karabowi @fuhlig1 @kresan Can you help here and shed some light on the purpose of `FairGenericStack::fFSTrackMap`, thx! The goal is to find readable names for variables denoting *what* this map maps onto *what*.

Reply:
From what I remember this map was introduced when we (@karabowi and @kresan) introduced the possibility to run `FastSimulation` in FairRoot. From what I remember the problem was the particle counting in the `FastSimulation` mode: normally, the secondaries created in the transport engine go to the end of the stack, while in the `FS` mode they were pushed just after the primary. That caused problem later on, so a map was created to track the `FS` secondaries.
After the event, the `FS` secondaries were to be moved to the end of the stack, and the points created by the had to be changed.

All that said. @fuhlig1 , @kresan , I think the `FastSim` mode was a dead end. `PANDA` is not using our implementation, they have their own. What about CBM or R3B? IMHO we could get rid of `FastSim` totally.
What do you think?

Reply:
> IMHO we could get rid of `FastSim` totally. What do you think?

Should we move that part of the discussion to a fresh issue?

Reply:
TBH, I don't know how to continue with this PR: If FastSim is a "dead end", then I probably should not invent new APIs like `FairGenericStack::FSTrackMapLookup` (which I currently do in this PR)?

But on the other hand, I am not feeling happy to just strip all FastSim stuff away.

If we wanted to merge this one "to simplify our current code", then I could already deprecated `FairGenericStack::FSTrackMapLookup`?

Reply:
Maybe we can discuss the issue on our meeting next Monday. Dima is pretty sure it is not used in `R3B`, `cbmroot` also has no traces of `FastSim`.

Reply:
Rethinking this, I would propose the following:

- We consider this PR independant of a possible FastSim deprecation / removal.
- Refactoring this might make the FastSim deprecation / removal even simpler, maybe
- And in a second step let's look at the FastSim stuff (please let's open an issue!)

Considering a possible deprecation / removal of these Fastsim things, I'd propose to keep the non-intuitive names for the time being.